### PR TITLE
Improve IllegalSignalError message

### PR DESCRIPTION
**Pull request by yogeshjain999**
This will generate below message when `IllegalSignalError` is raised -

```ruby
Trailblazer::Activity::Circuit::IllegalSignalError:
  Unrecognized Signal [true] returned from step <Dummy::Again> 
    Valid signals are: 
      #<Trailblazer::Activity::Railway::End::Failure semantic=:failure>
      #<Trailblazer::Activity::Railway::End::Success semantic=:success>
```

### CODE REVIEW

**Comment by apotonick** [comment]
Could we also have the throwing activity in the exception??

**Comment by yogeshjain999** [comment]
> Could we also have the throwing activity in the exception??

Good idea bro! Made both `task` and `signal` available in the exception.

File: test/circuit_test.rb [hunk 1]
```diff
@@ -95,4 +95,21 @@ class CircuitTest < Minitest::Spec
...
+    exception = assert_raises Trailblazer::Activity::Circuit::IllegalSignalError do
+      circuit = outer
+
+      # stub out fetching next task to return nothing and raise the exception
+      def circuit.next_for(task, signal); nil; end
+
+      circuit.([ {} ])
+    end
+
+    message = "Unrecognized Signal [\"to a\"] returned from step <#{Start}> \nValid signals are: \n\tto a\n\tto b"
```
Comment (apotonick) [inline test/circuit_test.rb]: It would be cool if it said "from `ActivityThrowingtheException`"?!?!

Comment (yogeshjain999) [reply test/circuit_test.rb]: Updated it @apotonick 👍
Is it cool if we use `exec_context` for getting actual activity ?

Comment (apotonick) [reply test/circuit_test.rb]: That's a great idea, didn't even think of it! :+1:

**Pull request merged**